graviti/schema/pyarrow.py

```diff
@@ -93,3 +95,30 @@ def __init__(
 
     def _to_pyobj(self) -> Dict[str, Any]:
         return {**super()._to_pyobj(), "url": self._url, "revision": self._revision}
+
+
+class FileArray(  # pylint: disable=too-few-public-methods
+    pa.ExtensionArray  # type: ignore[misc]
+):
+    """This class defines the PyArrow representation of FileArray."""
+
+    def __getitem__(self, index: Union[int, slice]) -> Union[File, pa.Array]:
+        if isinstance(index, slice):
+            return super().__getitem__(index)
```

Comment:
if input is a slice? what will be returned? `pa.Array` or `pa.ExtensionArray` or `pa.FileArray`?

Reply:
`pa.FileArray`

Reply:
should we update the return type to `pa.FileArray`?

Reply:
Also should we add `overload` for `__getitem__`